dna-src/zomes/my_zome/code/src/lib.rs

```diff
@@ -164,20 +164,44 @@ mod my_zome {
     }
 
     #[zome_fn("hc_public")]
-    fn create_goal(goal: Goal) -> ZomeApiResult<GetResponse<Goal>> {
+    fn create_goal(goal: Goal, parent_address: Option<Address>) -> ZomeApiResult<GetResponse<Goal>> {
```

Comment:
Maybe this function will have to return something like...

`ZomeApiResult<(GetResponse<Goal>, Option<GetResponse<Edge>>)>`

the () indicate a "tuple" in Rust... https://doc.rust-lang.org/rust-by-example/primitives/tuples.html

By adding these things into the response, we can also access them in the reducers in the UI